Library/Homebrew/extend/os/linux/linkage_checker.rb

```diff
@@ -80,9 +80,5 @@ def check_dylibs(rebuild_cache:)
     @unwanted_system_dylibs = @system_dylibs.reject do |s|
       SYSTEM_LIBRARY_ALLOWLIST.include? File.basename(s)
     end
-    # FIXME: Remove this when these dependencies are injected correctly (e.g. through `DependencyCollector`)
-    # See discussion at
-    #   https://github.com/Homebrew/brew/pull/13577
-    @undeclared_deps -= [CompilerSelector.preferred_gcc, "glibc", "gcc"]
```

Comment:
We might still need to keep `gcc` here, since everything build recently will prefer linkage with it to the host runtime GCC libraries.

Reply:
@carlocab can we remove for now and add back if 🔥?

Reply:
Sure, works for me.

Reply:
Adding this back: https://github.com/Homebrew/brew/pull/13796